File: java/1.java
import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++)
        {
            int comp = target - nums[i];
            if (map.containsKey(comp))
            {
                return new int[] {map.get(comp), i};
            }
            map.put(nums[i], i);
        }
        return new int[] {-1 , -1};
    };

    public static void main(String[] args) {
        Solution n1 = new Solution();
        int[] answer = n1.twoSum(new int[] {2,7,11,15},9);
        System.out.println("Sum is [" + answer[0] + "," + answer[1] + "]");
    };
}

File: python/1.py
## O(n^2) Method
# def twoSum( nums: list[int], target: int) -> list[int]:
#         for i in range(len(nums) - 1):
#             for j in range(len(nums) - 1 - i):
#                 if nums[j + 1 + i] == target - nums[i]:
#                     return [i, j + i + 1]
        
#         return []

## O(n) Method
def twoSum( nums: list[int], target: int) -> list[int]:
    hashTable = {}
    for i in range(len(nums)):
        if nums[i] in hashTable.values():
            return [(list(hashTable.values())).index(nums[i]), i]
        else:
            hashTable[i] = target - nums[i]
    print(hashTable)
    return []

print(twoSum([2,7,11,15], 9))
print(twoSum([3,2,4], 6))
print(twoSum([3,3], 6))
print(twoSum([2,5,5,11], 10))